### ISSUE TITLE
Cannot compile batteries on ocaml multicore 4.12.0 + domains

### ISSUE THREAD

**Issue by Termina1** [ocamllabs/ocaml-multicore#625]
I see this issue (https://github.com/ocaml-batteries-team/batteries-included/issues/949), but I still can't compile battries.3.30
```
#=== ERROR while compiling batteries.3.3.0 ====================================#
# context     2.0.8 | macos/x86_64 | ocaml-variants.4.12.0+domains | https://opam.ocaml.org#d002182b
# path        ~/.opam/4.12.0+domains/.opam-switch/build/batteries.3.3.0
# command     ~/.opam/opam-init/hooks/sandbox.sh build make all
# exit-code   2
# env-file    ~/.opam/log/batteries-43556-a71bb4.env
# output-file ~/.opam/log/batteries-43556-a71bb4.out
### output ###
# [...]
# Error: The implementation src/batGc.ml
#        does not match the interface src/batGc.cmi:
#        The value `huge_fallback_count' is required but not provided
#        File "src/batGc.mliv", line 289, characters 0-74: Expected declaration
#        The value `get_credit' is required but not provided
#        File "src/batGc.mliv", line 283, characters 0-71: Expected declaration
#        The value `get_bucket' is required but not provided
#        File "src/batGc.mliv", line 276, characters 0-70: Expected declaration
#        The value `get_minor_free' is required but not provided
#        File "src/batGc.mliv", line 270, characters 0-61: Expected declaration
# Command exited with code 2.
# make: *** [all] Error 10
```

**Comment by ctk21** [comment]
This may be due to a recent change to clean up the version returned by the multicore compiler. 

The multicore compiler used to return `4.12+multicore` for versions including `+domains` and `+domains+effects`. This was changed recently with the merge of #514 (see also discussion on #606) and now the version string will be `4.12+domains` and `4.12+domains+effects` explicitly matching the github branches and features.

I think batteries will need to update the string matching in `prefilter.ml` 
https://github.com/ocaml-batteries-team/batteries-included/blob/master/build/prefilter.ml#L55-L56

**Comment by Sudha247** [comment]
Fixed by https://github.com/ocaml-batteries-team/batteries-included/pull/1044
